poly-synth/src/poly-synth-voice.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+class PolySynthVoice {
+  /**
+   * @class  PolySynthVoice
+   * @constructor
+   * A voice generates a waveform and filters it, exposing the sound through
+   * this.output.
+   * @param {AudioContext} context the audio context
+   * @param {String} note the note corresponding to the frequency
+   * @param {Number} frequency the frequency corresponding to the voice's note
+   * @param {Number} cutoff the cutoff frequency for the lowpass filter
+   * @param {PolySynth} parentSynthesizer the synthesizer that manages this
+   *                                      voice.
+   * @param {Object} options optional parameters
+   * @param {String} options.waveform the oscillator's waveform
+   * @param {Number} options.attack milliseconds until full amplitude
+   * @param {Number} options.decay milliseconds until sustain level
+   * @param {Number} options.sustain the steady amplitude of the note as it is
+   *                                 pressed
+   * @param {Number} options.release milliseconds between the release of a note
+   *                                 and zero amplitude
+   */
+  constructor(context, note, frequency, cutoff, parentSynthesizer, options) {
+    if (!(context instanceof AudioContext))
```

Comment:
I don't think this check is needed. `PolySynthVoice` can only be used within `PolySynth`, so the parent object will be no-op with an invalid context.